Update test to use filename generated instead of key name as what loo…

Coderrob opened the pull request:
- Update test to use filename generated instead of key name as what looked intended.
- Update MimeUtility to handle null file variable. 
   - Assuming return default instead of possible null reference exception or throwing argument null exception is intended by author.

zone117x commented on MimeUtility.cs:
```diff
@@ -6,41 +6,31 @@
+        ///     Dictionary of all available types (lazy loaded on first call)
         /// </summary>
-        public static ReadOnlyDictionary<string, string> TypeMap => _lazyDict.Value;
+        public static ReadOnlyDictionary<string, string> TypeMap => LazyDict.Value;
```
Can the whitespace and stylist changes be reverted?

Coderrob replied:
Can do!

zone117x commented on MimeUtility.cs:
```diff
@@ -6,41 +6,31 @@
-                fileExtension = file;
-            }
-            string mimeType = KnownMimeTypes.LookupType(fileExtension.ToLowerInvariant());
+            if (string.IsNullOrEmpty(file)) return UnknownMimeType;
```
Considering typical use cases, I think empty strings can be expected, and returning `UnknownMimeType` immediately makes sense. But for null I think we should be throwing an argument null exception.

Coderrob replied:
I was thinking that as well. I originally put:
```
if (string.IsNullOrEmpty(file)) throw new ArgumentNullException(nameof(file));
```
That seems reasonable? 

Thanks for putting together this package. I was previously using the AspnetCore StaticFiles package just to use `FileExtensionContentTypeProvider` to do the same thing your dictionary provided.

zone117x replied:
That null check and exception look good. 

I imagine many of the file inputs come from arbitrary web clients, and I think developers would be more surprised by an empty string throwing and exception rather than returning unknown.

zone117x commented on Test/BasicTest.cs:
```diff
@@ -1,32 +1,33 @@
             {
-                var foundType = MimeUtility.GetMimeMapping(t.Key);
+                var filePath = Path.Combine(Path.GetTempPath(), "examplefile." + t.Key);
+                var foundType = MimeUtility.GetMimeMapping(filePath);
```
Nice catch here -- the `filePath` was never being used. Can you perform this fix within the `TestCommonTypesAsFilePaths` method? This `TestCommonTypes` is intended to test bare extension strings without the dot.

Coderrob commented:
@zone117x sorry for the delay. Updated the style changes back to the original format. I updated the guards for the file argument to do a null check, and then an empty string check. Afterwards I updated the unit tests to correct the one you'd mentioned, and added a few more to cover the behavior changes of the argument checks.

zone117x commented:
Looks great, thank you!
